enzyme/test/Integration/ReverseMode/cin.cpp

```diff
@@ -0,0 +1,40 @@
+// RUN: %clang++ -std=c++11 -fno-exceptions -ffast-math -O0 %s -S -emit-llvm -o - | %opt - %loadEnzyme -enzyme -S
+// RUN: %clang++ -std=c++11 -fno-exceptions -ffast-math -O1 %s -S -emit-llvm -o - | %opt - %loadEnzyme -enzyme -S
```

Comment:
You may be able to use `https://en.cppreference.com/w/cpp/io/basic_istream/swap` to swap cin with a stringbuffer stream which you have the input you want already prepared

Reply:
done